pkg/api/types.go

```diff
@@ -2703,6 +2703,21 @@ type ServiceSpec struct {
 	// and ExternalTrafficPolicy is set to Local.
 	// +optional
 	HealthCheckNodePort int32
+
+	// TolerateUnreadyEndpoints, when set to true, indicates that endpoints
+	// controller should create endpoints for the unready Pods associated with
+	// the Service. This is useful for allowing the Pods associated with a
+	// headless Service to be DNS resolvable without respect to their readiness checks.
+	// Endpoints of these Services retain their DNS records and continue
+	// receiving traffic for the Service from the moment the kubelet starts all
+	// containers in the Pod and marks it "Running", until the kubelet stops all
```

Comment:
> In reality it causes the endpoints controller to completely ignore readiness. 

Implementation is what you just described. It completely ignores pod readiness. So no matter what state pod is in, service is not affected. I hope we can change the documentation as well, since I believe the init container case is a valid one.